### PR TITLE
Stop deprecation warnings on arbitrary SCRAPY-prefixed env vars

### LINKED ISSUE #4374: "DeprecationWarning: Use of 'SCRAPY_'-prefixed environment variables" always shows
`ScrapyDeprecationWarning: Use of 'SCRAPY_'-prefixed environment variables to override settings is deprecated.`

This deprecation warning can't be shut up inside a project.

https://github.com/scrapy/scrapy/blob/034e2c31c7d55333c3de208f80dcee1bf45ef9b9/scrapy/utils/project.py#L17-L19

This is always true, if inside a project. Because of
https://github.com/scrapy/scrapy/blob/034e2c31c7d55333c3de208f80dcee1bf45ef9b9/scrapy/utils/conf.py#L77-L84
We should probably stop "emulating" ENV vars there, and pass this somehow differently.

Additionally, the `ScrapyDeprecationWarning` for `env_overrides` is too generic: It never tells you _which_ variable(s) are actually set. This is plain annoying. Please make it more specific.

### PR DESCRIPTION
**Pull request by Gallaecio**
Noticed by @nyov.

I’m thinking that this is technically backward-incompatible. Before, we were actually defining a `SETTINGS_MODULE` setting, and potentially other variables. Should I just whitelist warning-wise but still allow the setting to be defined?

Do we issue a specific warning about that? i.e. If a setting named after one of those white-listed variables is read from `Settings`, warn about it.

To-Do:

- [x] Indicate the offending variables, as requested by @nyov 

Fixes #4374

### CODE REVIEW

**Comment by Gallaecio** [comment]
Adding to the 2.0 milestone, given #4300 was merged after 1.8.

File: scrapy/utils/project.py [hunk 1]
```diff
@@ -75,8 +75,15 @@ def get_project_settings():
...
+        'SCRAPY_PICKLED_SETTINGS_TO_OVERRIDE',  # Already warned above
+        'SCRAPY_PROJECT',
+        'SCRAPY_PYTHON_SHELL',
+        'SCRAPY_SETTINGS_MODULE',
+    }
     env_overrides = {k[7:]: v for k, v in os.environ.items() if
-                     k.startswith('SCRAPY_')}
+                     k.startswith('SCRAPY_') and k not in whitelist}
     if env_overrides:
         warnings.warn("Use of 'SCRAPY_'-prefixed environment variables to override settings is deprecated.", ScrapyDeprecationWarning)
```
Comment (nyov) [inline scrapy/utils/project.py]: How about
```diff
     if env_overrides:
-        warnings.warn("Use of 'SCRAPY_'-prefixed environment variables to override settings is deprecated.", ScrapyDeprecationWarning)
+        overridden = ', '.join(env_overrides)
+        warnings.warn("Use of 'SCRAPY_'-prefixed environment variables to override settings is deprecated: {}".format(overridden), ScrapyDeprecationWarning)
         settings.setdict(env_overrides, priority='project')
```

**Comment by Gallaecio** [comment]
I’ve changed the code based on @nyov’s suggestion and mine about warning about the reading of unintended settings.

Tomorrow I’ll work on test coverage.

**Comment by wRAR** [comment]
No f-strings in 3.5 :)

**Comment by Gallaecio** [comment]
I’ve simplified the code to just:
- Stop warning about valid environment variables.
- List offending environment variables when warning about them.

**Pull request merged**